btr-api/sample_data/generate_data.py

```diff
@@ -0,0 +1,212 @@
+
+    for _ in range(num):
+        name = {
+            'type': fake.random_element(['legal', 'knownAs', 'previous']),
+            'fullName': fake.name(),
+            'familyName': fake.last_name(),
+            'givenName': fake.first_name()
+        }
+
+        address = {
+            'type': fake.random_element(['legal', 'knownAs', 'previous']),
+            'fullName': fake.name(),
+            'familyName': fake.last_name(),
+            'givenName': fake.first_name()
+        }
+
+        random_country = fake.random_element(pycountry.countries)
+        nationality = {
+            'code': random_country.alpha_2,
+            'name': random_country.name
+        }
+
+        person_stmnts.append({
+            'statementID': 'sample-person-' + fake.uuid4(),
+            'statementType': fake.random_element(['individual', 'organization']),
+            'statementDate': statement_date,
+            'personType': 'knownPerson',
+            'isComponent': fake.random_element([True, False]),
+            'publicationDetails': {
+                "publicationDate":  statement_date,
+                "bodsVersion": "1.0",
+                "publisher": {
+                    "uri": "http://publisher.uri",
+                    "name": "Publisher Name",
+                    "source": {
+                    "url": "http://source.url"
+                    }
+                }
+            },
+            'names': name,
+            'nationalities': [nationality],
+            'identifiers': [
+                # NOTE: fake identifiers + tax numbers based on _link
+                {'id': f"TST-P{fake.uuid4()}", 'scheme': 'TST-P'},
+                {'id': f"TST-TAX{fake.uuid4()}", 'scheme': 'TST-TAX'},
+            ],
+            
+            'birthDate': statement_date,
+            'addresses': [address],
+            'taxResidencies': [nationality],
+            'email': get_random_email(),
+            'hasTaxNumber': fake.random_element([True, False])
+        })
+
+    return person_stmnts
+
+
+def _get_filing(entity_stmnt: dict, person_stmnts: dict, ooc_stmnts: list):
+    """Return the filing based on the given statements."""
+    filings = {
+        'businessIdentifier': 'sample-business-' + fake.uuid4(),
+        'effectiveDate': entity_stmnt['statementDate'],
+        'entityStatement': entity_stmnt,
+        'personStatements': person_stmnts,
+        'ownershipOrControlStatements': ooc_stmnts
+    }
+    return filings
+
+
+def generate_data():
+    """Generate sample data for search."""
+    user = User.find_by_username('service-account-nds')
+
+    if not user:
+        current_app.logger.debug('error user not found.')
+        return
+    
+    global fake
+
+    for locale, settings in SETTING.items():
+        fake = Faker(locale)
+        for _ in range(settings[0]):
+            entity_stmnt = _get_entity_stmnt(get_fake_date())
+            person_stmnts = _get_person_stmnts(get_fake_date(), settings[1])
```

Comment:
The statement date for this needs to be the same as or before the ooc statement date so I would just pass the same date in that you pass in for the entity statement. Otherwise the generated record might not make sense and cause us issues later